### PR TITLE
Update test framework versions and fix issues with tests

### PR DESCRIPTION
**Pull request by natemcmaster**
Update to xunit 2.3.0-beta2 and vstest 15.3

Changes:
 - no more guids thanks to VS 15.3
 - Fix issues in tests as identified by xunit.analyzers
 - Remove duplicate test data
 - Add a missing IDisposable so test cleanup happens properly

### CODE REVIEW

File: test/Microsoft.AspNetCore.Server.Kestrel.Core.Tests/FrameRequestHeadersTests.cs [hunk 1]
```diff
@@ -30,7 +30,6 @@ public void SettingUnknownHeadersWorks()
 
             headers["custom"] = new[] { "value" };
 
-            Assert.NotNull(headers["custom"]);
```
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Server.Kestrel.Core.Tests/FrameRequestHeadersTests.cs]: FYI, this returned `StringValues` which is a value type (aka can never be null).

**Comment by natemcmaster** [comment]
`ResponseTests.FailedWritesResultInAbortedRequest` hangs indefinitely when upgrading to xunit 2.3.0. Any ideas @cesarbs ?

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs [hunk 2]
```diff
@@ -2118,12 +2115,12 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
                     // Don't wait to receive the response. Just close the socket.
                 }
 
-                connectionCloseWh.Set();
+                connectionCloseWh.TrySetResult(null);
 
                 // Write failed
-                await Assert.ThrowsAsync<TaskCanceledException>(async () => await writeTcs.Task);
+                await Assert.ThrowsAsync<TaskCanceledException>(async () => await writeTcs.Task).TimeoutAfter(TimeSpan.FromSeconds(15));
```
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs]: This times out.

File: test/Microsoft.AspNetCore.Server.Kestrel.Core.Tests/FrameHeadersTests.cs [hunk 3]
```diff
@@ -171,12 +161,9 @@ public void TestParseConnectionMultipleValues(string value1, string value2, Conn
         [InlineData(" ,,", TransferCoding.None)]
         [InlineData(",, ", TransferCoding.None)]
         [InlineData(" , ,", TransferCoding.None)]
-        [InlineData(" , ,", TransferCoding.None)]
         [InlineData(" , , ", TransferCoding.None)]
         [InlineData("chunked,", TransferCoding.Chunked)]
         [InlineData("chunked,,", TransferCoding.Chunked)]
-        [InlineData(",chunked", TransferCoding.Chunked)]
-        [InlineData(",,chunked", TransferCoding.Chunked)]
```
Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.Kestrel.Core.Tests/FrameHeadersTests.cs]: Why did you remove some of these?

File: test/Microsoft.AspNetCore.Server.Kestrel.Core.Tests/FrameResponseHeadersTests.cs [hunk 4]
```diff
@@ -66,7 +66,6 @@ public void InitialDictionaryIsEmpty()
         [InlineData("Ser™ver", "Data")]
         [InlineData("Server", "Da™ta")]
         [InlineData("Unknown™-Header", "Data")]
-        [InlineData("Ser™ver", "Data")]
```
Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.Kestrel.Core.Tests/FrameResponseHeadersTests.cs]: ?

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs [hunk 5]
```diff
@@ -1160,8 +1158,6 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
         [Theory]
         [InlineData("gzip")]
         [InlineData("chunked, gzip")]
-        [InlineData("gzip")]
-        [InlineData("chunked, gzip")]
```
Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs]: lol 👍

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs [hunk 6]
```diff
@@ -2075,27 +2072,27 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
             var largeString = new string('a', maxBytesPreCompleted + 1);
 
             var writeTcs = new TaskCompletionSource<object>();
-            var registrationWh = new ManualResetEventSlim();
-            var connectionCloseWh = new ManualResetEventSlim();
+            var registrationWh = new TaskCompletionSource<object>();
+            var connectionCloseWh = new TaskCompletionSource<object>();
```
Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs]: Why?

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs (hunk 2 above)
Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs]: Could be the switch to tcs. The continuation runs as soon as you call `SetResult` (i.e. in the same thread). We've had test deadlocks because of that before.

File: test/Microsoft.AspNetCore.Server.Kestrel.Core.Tests/FrameHeadersTests.cs (hunk 3 above)
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Server.Kestrel.Core.Tests/FrameHeadersTests.cs]: They are duplicates. See 6 lines down.

File: test/Microsoft.AspNetCore.Server.Kestrel.Core.Tests/FrameResponseHeadersTests.cs (hunk 4 above)
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Server.Kestrel.Core.Tests/FrameResponseHeadersTests.cs]: Also a duplicate. See 3 lines up.

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs (hunk 6 above)
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs]: The test was hanging indefinitely on `.Wait()`. I switched to tcs so I could add .TimeoutAfter assertions.

**Comment by natemcmaster** [comment]
I figured out the hanging test. It was a race condition between the response starting and the connection closing. Test depended on the connection remaining open until the response started.

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs (hunk 6 above)
Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs]: Rename the vars if you're keeping the TCSs.

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs [hunk 7]
```diff
@@ -2115,15 +2114,16 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
...
 
-                connectionCloseWh.Set();
+                connectionCloseWh.TrySetResult(null);
 
                 // Write failed
-                await Assert.ThrowsAsync<TaskCanceledException>(async () => await writeTcs.Task);
+                await Assert.ThrowsAsync<TaskCanceledException>(async () => await writeTcs.Task).TimeoutAfter(TimeSpan.FromSeconds(15));
                 // RequestAborted tripped
-                Assert.True(registrationWh.Wait(1000));
+                Assert.True(registrationWh.Task.IsCompleted);
```
Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs]: await with timeout

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs (hunk 2 above)
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs]: I decided to switch back to original code (non tcs version) and just add another ManualResetEventSlim to fix the race condition.

**Comment by natemcmaster** [comment]
🆙 📅

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs [hunk 8]
```diff
@@ -2077,17 +2074,19 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
...
+                requestStartWh.Set();
                 var response = httpContext.Response;
                 var request = httpContext.Request;
                 var lifetime = httpContext.Features.Get<IHttpRequestLifetimeFeature>();
 
                 lifetime.RequestAborted.Register(() => registrationWh.Set());
 
                 await request.Body.CopyToAsync(Stream.Null);
-                connectionCloseWh.Wait();
+                connectionCloseWh.Wait(1000);
```
Comment (halter73) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs]: Can we add an `Assert.True` here?

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs [hunk 9]
```diff
@@ -2115,13 +2114,14 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
                         "Content-Length: 5",
                         "",
                         "Hello");
-                    // Don't wait to receive the response. Just close the socket.
+
+                    requestStartWh.Wait(1000);
```
Comment (halter73) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs]: ditto

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs (hunk 8 above)
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs]: Updated.
